IHP/ValidationSupport/ValidateIsUnique.hs

```diff
@@ -52,3 +52,44 @@ validateIsUnique fieldProxy model = do
         Just value | not $ (getField @"id" model) == (getField @"id" value) -> pure (attachValidatorResult fieldProxy (Failure "This is already in use") model)
         _ -> pure (attachValidatorResult fieldProxy Success model)
 {-# INLINE validateIsUnique #-}
+
+
+-- | Overrides the error message of a given IO validator function.
+--
+-- __Example:__ Validate that an email is unique with a custom error message
+--
+-- > action CreateUserAction = do
+-- >     let user = newRecord @User
+-- >     user
+-- >         |> fill @'["email"]
+-- >         |> withCustomErrorMessage' "Email Has Already Been Used" validateIsUnique #email
```

Comment:
Can we call this function `withCustomErrorMessageIO`? Also could you add a note to https://ihp.digitallyinduced.com/Guide/validation.html#customizing-error-messages ?